hosted/gke/k8s_chart_support/k8s_chart_support_suite_test.go

```diff
@@ -0,0 +1,240 @@
+	zone                    = helpers.GetGKEZone()
+	project                 = helpers.GetGKEProjectID()
+	rancherVersion          = os.Getenv("RANCHER_VERSION")
+	rancherUpgradedVersion  = os.Getenv("RANCHER_UPGRADE_VERSION")
+	k3sKubeconfig           = os.Getenv("KUBECONFIG")
+	k8sUpgradedMinorVersion = os.Getenv("K8S_UPGRADE_MINOR_VERSION")
+)
+
+func TestK8sChartSupport(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "K8sChartSupport Suite")
+}
+
+var _ = BeforeEach(func() {
+	Expect(rancherVersion).ToNot(BeEmpty())
+	// For upgrade tests, the rancher version should not be an unreleased version (for e.g. 2.8-head); ideally it should be a released version
+	Expect(rancherVersion).ToNot(ContainSubstring("head"))
+
+	Expect(rancherUpgradedVersion).ToNot(BeEmpty())
+	Expect(k8sUpgradedMinorVersion).ToNot(BeEmpty())
+	Expect(k3sKubeconfig).ToNot(BeEmpty())
+
+	By("Install Rancher Manager to the base version if not already installed and uninstall the operator charts if existing", func() {
+
+		if currentVersion := helpers.GetRancherVersion(); currentVersion.Devel {
+			helpers.DeployRancherManager("latest", rancherVersion, true)
+		}
+
+		charts := helpers.ListOperatorChart()
+		for _, chart := range charts {
+			args := []string{"uninstall", chart.Name, "--namespace", helpers.CattleSystemNS}
+			err := kubectl.RunHelmBinaryWithCustomErr(args...)
+			Expect(err).To(BeNil())
+		}
+	})
+
+	var err error
+	ctx, err = helpers.CommonBeforeSuite("gke")
+	Expect(err).To(BeNil())
+	clusterName = namegen.AppendRandomString("gkehostcluster")
+
+	k8sVersion, err = helper.DefaultGKE(ctx.RancherClient, project, ctx.CloudCred.ID, zone, "")
+	Expect(err).To(BeNil())
+	GinkgoLogr.Info(fmt.Sprintf("Using GKE version %s", k8sVersion))
+
+})
+
+func commonchecks(ctx *helpers.Context, cluster *management.Cluster, clusterName, rancherUpgradedVersion, hostname, k8sUpgradedVersion string) {
+	var newChartVersion, oldChartVersion string
+
+	By("checking cluster name is same", func() {
+		Expect(cluster.Name).To(BeEquivalentTo(clusterName))
+	})
+
+	By("checking service account token secret", func() {
+		success, err := clusters.CheckServiceAccountTokenSecret(ctx.RancherClient, clusterName)
+		Expect(err).To(BeNil())
+		Expect(success).To(BeTrue())
+	})
+
+	By("checking all management nodes are ready", func() {
+		err := nodestat.AllManagementNodeReady(ctx.RancherClient, cluster.ID, helpers.Timeout)
+		Expect(err).To(BeNil())
+	})
+
+	By("checking all pods are ready", func() {
+		podErrors := pods.StatusPods(ctx.RancherClient, cluster.ID)
+		Expect(podErrors).To(BeEmpty())
+	})
+
+	By("checking the chart version", func() {
+		oldCharts := helpers.ListOperatorChart()
+		oldChartVersion = oldCharts[0].DerivedVersion
+		Expect(oldChartVersion).ToNot(BeEmpty())
+	})
+
+	By("upgrading rancher", func() {
+		helpers.DeployRancherManager("latest", rancherUpgradedVersion, true)
+
+		By("regenerating the token and initiating a new rancher client", func() {
```

Comment:
_helpers.CommonBeforeSuite_ could be used here ? It will only create extra Cloud creds IMO